src/runAll.js

```diff
@@ -62,7 +62,16 @@ module.exports = function runAll(config) {
       renderer
     }
 
-    if (tasks.length) {
+    // If all of the configured "linters" should be skipped
+    // avoid executing any "stashing changes..." logic
+    let isSkippingAllLinters = true
+    tasks.forEach(task => {
+      if (!task.skip()) {
```

Comment:
I’m still wondering why do we need to execute a function defined on each task here. Also I’d prefer a more functional code style but consider this a nit pick

Reply:
Say I've got two rules for two different tasks/linters, ie. one for TypeScript and one for SASS files.

```
"*.ts": ["tslint --fix", "git add"],
"*.scss": ["sass-lint --fix", "git add"]
```

I want to ensure I'm checking all tasks for file changes as somebody may be only committing TypeScript files or SASS files but not both.

So if only TypeScript files are staged but not SASS files, the first task will not skip but the second task will.
If SASS files are staged but not TypeScript files, the first task WILL skip but not the second.

Does that make sense?

Reply:
Yes I think I’ve got the logic. Let’s rewrite this in a more way so it matches the code style. What about using `some` or `every` of Array?

Reply:
Alright :)
I'm personally not a fan of functional programming paradigms applied to arrays, but I'll make the change regardless!